tests/test_subprocess_runner.py

```
@@ -863,3 +864,311 @@ def test_stall_timeout_is_clamped_to_the_sampling_floor(tmp_path, monkeypatch):
+        probed_on.append(threading.current_thread().name)
+        return [str(p) for p in (base, *parts)]
+
+    async def _go():
+        return await runner_module.remove_partial_output(discover=_discover)
+
+    assert asyncio.run(_go()) is True
+    assert not any(p.exists() for p in (base, *parts))
+    assert probed_on == ["fs-probe"]  # not the event loop's thread
+
+
+def test_remove_partial_output_bounds_a_wedged_discovery(monkeypatch):
```

Ruff (0.16.1)

[warning] Unused function argument: `monkeypatch`

(ARG001)